src/settings.js

```diff
@@ -1982,6 +1982,10 @@ var TRUSTED_TYPES = 0;
 // settings is *only* needed when also explicitly targeting older browsers.
 var POLYFILL = 1;
 
+// The parameters used when doing a fetch() of data over the network, like of
+// the wasm file.
+var FETCH_PARAMS = "{ credentials: 'same-origin' }";
```

Comment:
I'm not a fan of adding settings like this that take arbitrary strings because it makes command line parsing tricky.

Would a Module override not make more sense here?

IIRC we had another issue regarding fetch settings where we were discussing how to best add a Module override without incurring too much cost?

Reply:
One can use a file to avoid shell parsing. Since we have that, I think it's ok to add settings like this.

A module override is another option, but it will increase code size for all users, which this avoids.

Not sure what other issue you refer to, but that does sound a little familiar.

Reply:
But can't we make the Module overide optional?  I.e. opt-in via INCOMING_MODULE_JS_API?

Reply:
Ah, I found it: https://github.com/emscripten-core/emscripten/pull/15569

Reply:
So far we don't have any settings for which "inline JavaScript code" is the content of the setting (at least not by default or in common usage).  I'd rather not start having JavaScript snipped in settings.

Reply:
There is precedent, at least `PTHREAD_POOL_SIZE` is a setting that receives a string that is placed into the output, like this one. A common use case, mentioned in the docs there, is to do this: `-s PTHREAD_POOL_SIZE=navigator.hardwareConcurrency`. That JS will automatically make use use a pool of the number of cores in the machine.

But, yes, I see this is a dupe of the other issue, so let's close this and focus there...

Reply:
Ah yes, I knew there something like `PTHREAD_POOL_SIZE=navigator.hardwareConcurrency` out there... still its quite limited.. and I hope we can avoid adding more of it.. and I hope nobody is doing horrible things such as passing more complicated JS expressions.  Most user of PTHREAD_POOL_SIZE are still passing integers I think/hope.

Reply:
Why is it horrible, though? If you use a response file, it's in a separate file. Seems useful to me to have JS customization hooks..?

Reply:
For one thing, we already have JS customization hooks in the form of incoming Module attributes.  Why invent a second way inject customization when we already have one?   Especially one that involves one new setting for each customization point.

Reply:
The main reason is code size. Each new Module attribute is a cost, unless we want to start adding new Module attributes that are opt-in - but that also means some form of new setting. As I said in the other issue, though, we could maybe add a general option to get a list of all new opt-int Module attributes?

Reply:
Yes, I think opt-in Module attributes make sense for this kind of thing.